File: .py
# Temel Person sınıfı
class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def display_info(self):
        return f"Ad: {self.name}, Yaş: {self.age}"

# Student sınıfı
class Student(Person):
    def __init__(self, name, age, student_id):
        super().__init__(name, age)
        self.student_id = student_id

    def display_info(self):
        return f"{super().display_info()}, Öğrenci No: {self.student_id}"

# Teacher sınıfı
class Teacher(Person):
    def __init__(self, name, age, subject):
        super().__init__(name, age)
        self.subject = subject

    def display_info(self):
        return f"{super().display_info()}, Ders: {self.subject}"

# Ana Program
def main():
    # Öğrenci örneği oluşturma
    student = Student("Seydi Emir", 20, "12345")
    print("Öğrenci Bilgileri:", student.display_info())

    # Öğretmen örneği oluşturma
    teacher = Teacher("Elif Hoca", 40, "Matematik")
    print("Öğretmen Bilgileri:", teacher.display_info())

if __name__ == "__main__":
    main()

File: .cpp
#include <iostream>
#include <string>
using namespace std;

// Temel Person sınıfı
class Person {
protected:
    string name;
    int age;

public:
    Person(string name, int age) : name(name), age(age) {}

    virtual string displayInfo() {
        return "Ad: " + name + ", Yaş: " + to_string(age);
    }
};

// Student sınıfı
class Student : public Person {
private:
    string studentID;

public:
    Student(string name, int age, string studentID) : Person(name, age), studentID(studentID) {}

    string displayInfo() override {
        return Person::displayInfo() + ", Öğrenci No: " + studentID;
    }
};

// Teacher sınıfı
class Teacher : public Person {
private:
    string subject;

public:
    Teacher(string name, int age, string subject) : Person(name, age), subject(subject) {}

    string displayInfo() override {
        return Person::displayInfo() + ", Ders: " + subject;
    }
};

// Ana Program
int main() {
    // Öğrenci örneği oluşturma
    Student student("Seydi Emir", 20, "12345");
    cout << "Öğrenci Bilgileri: " << student.displayInfo() << endl;

    // Öğretmen örneği oluşturma
    Teacher teacher("Elif Hoca", 40, "Matematik");
    cout << "Öğretmen Bilgileri: " << teacher.displayInfo() << endl;

    return 0;
}
